server/app/controllers/admin/NotChangeableException.java

```diff
@@ -0,0 +1,8 @@
+package controllers.admin;
+
+/** Exception when a request is sent to modify something that can not be changed. */
+class NotChangeableException extends RuntimeException {
+  public NotChangeableException(String message) {
+    super(message);
+  }
+}
```

Comment:
issue (blocking): since these aren't caught and are runtime exceptions, they'll result in a 500 status code which is incorrect -- the client is sending a bad request and should receive a 400.

It's annoying since there are so many places where this is checked. I wonder if an alternative could be to not extend RuntimeException and extend the [the default error handler](https://www.playframework.com/documentation/2.8.x/JavaErrorHandling) to handle  `NotChangeableException`

Reply:
My plan was to go down that route or piggy back on any changes Brett made subsequently.  Blocking these invalid edits seemed worth getting in.  Any pre-existing knowledge on the level of effort on doing the play handling for these?

Reply:
> Any pre-existing knowledge on the level of effort on doing the play handling for these?

nope, my guess is it'd be fairly quick though. you might be able to get away with leaving `NotChangeableException` as a runtime exception, [adding a new singleton error handler that extends the default one](https://www.playframework.com/documentation/2.8.x/JavaErrorHandling#Extending-the-default-error-handler), and then overriding `onServerError` that checks if  the throwable is a `NotChangeableException` and doing `return Results.badRequest("form submitted with stale data");` if it is.

if that inflates the level of effort more than expected i'm ok punting though.

Reply:
Done!